simplesat/rules_generator.py

```diff
@@ -399,6 +406,49 @@ def key(package):
         )
         self._add_rule(rule, "job")
 
+    def _add_constrain_job_rules(self, job):
+        """
+        Add ad-hoc rules that constrain the solution to satisfy the
+        requirement.
+
+        This operation differs from just adding an "install" rule in that that
+        it does *not* force installation of any packages. I.e. the solution
+        must *not* satisfy the *negation* of the requirement given.
+
+        For example the install job for "MKL > 10.0" would translate to "Must
+        install an MKL version greater than 10.0.0-0," whereas a constrain job
+        would be read as "Must not install an MKL version less than or equal to
+        10.0.0-0."
+
+        """
+        # XXX: an alternative approach might be to actually negate the
+        # constraint, but that requires more logic and seems more likely to
+        # need special rules for edge cases.
```

Comment:
Actually maybe it wouldn't be much harder? `pool.what_provides(R(job.requirement).negated())` isn't so crazy. Would that work in all cases?